src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,68 @@
+package mate.academy.bookstore.repository.impl;
+
+import jakarta.persistence.EntityManagerFactory;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.bookstore.exception.EntityNotFoundException;
+import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.repository.BookRepository;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(EntityManagerFactory entityManagerFactory) {
+        sessionFactory = entityManagerFactory.unwrap(SessionFactory.class);
+    }
+
+    @Override
+    public Book save(Book book) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't save book: " + book + " to the DB", e);
```

Comment:
```suggestion
            throw new DataProcessingException("Can't save book: " + book + " to the DB", e);
```

same for other places